### PR TITLE
Add data to Uri checks

### PR DESCRIPTION
**Pull request by gabrielemilan**
**What this PR does / why we need it**:
Add response information in data property of payload

**Which issue(s) this PR fixes**:
none

Please reference the issue this PR will close: #_[issue number]_

**Special notes for your reviewer**:
If you think is good, I can add some tests.
Thanks.

**Does this PR introduce a user-facing change?**:

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [x] Code compiles correctly
- [x] Created/updated tests
- [x] Unit tests passing
- [x] End-to-end tests passing
- [ ] Extended the documentation
- [x] Provided sample for the feature

### CODE REVIEW

**Comment by CarlosLanderas** [comment]
Hello @gabrielemilan. Thanks four your PR.

In this PR nothing is configurable, you are not allowing the user to configure the data. It might be better to allow configuring an HttpContent. You also need to take in consideration the HttpVerb. It can be a GET request...

Thi kind of feature should extend the tests as well..

**Comment by gabrielemilan** [comment]
Added configurations and tests

File: src/HealthChecks.Uris/UriHealthCheckOptions.cs [hunk 1]
```diff
@@ -27,7 +27,7 @@ public class UriOptions : IUriOptions
         private readonly List<(string Name, string Value)> _headers = new List<(string Name, string Value)>();
 
         internal IEnumerable<(string Name, string Value)> Headers => _headers;
-
+
```
Comment (tomatualus) [inline src/HealthChecks.Uris/UriHealthCheckOptions.cs:30]: Remove whitespace

File: test/FunctionalTests/HealthChecks.Uris/StringAssertionExentions.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+using FluentAssertions.Primitives;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace FunctionalTests.HealthChecks.Uris
+{
+    public static class StringAssertionExentions
+    {
+        public static AndConstraint<StringAssertions> ContainCheckAndResult(this StringAssertions assertions, string url, int statusCode, string reason, string body, string verb)
+        {
+
```
Comment (tomatualus) [inline test/FunctionalTests/HealthChecks.Uris/StringAssertionExentions.cs:11]: remove blank line

File: test/FunctionalTests/HealthChecks.Uris/UriHealthCheckTests.cs [hunk 3]
```diff
@@ -267,5 +269,91 @@ public async Task be_healthy_if_request_sucess_and_timeout_is_configured()
...
+              .Configure(app =>
+              {
+                  app.UseHealthChecks("/health", new HealthCheckOptions()
+                  {
+                      Predicate = r => r.Tags.Contains("uris"),
+                      ResponseWriter = UIResponseWriter.WriteHealthCheckUIResponse
+                  });
+              });
+
+            var server = new TestServer(webHostBuilder);
```
Comment (tomatualus) [inline test/FunctionalTests/HealthChecks.Uris/UriHealthCheckTests.cs:305]: The whole TestServer and WebHostBuilder code could be refactored into a function to reduce code duplication.

Comment (tomatualus) [reply test/FunctionalTests/HealthChecks.Uris/UriHealthCheckTests.cs:305]: The same code appears in `be_unhealthy_if_request_success_and_response_data_is_configured` unit test

**Comment by sungam3r** [comment]
Rel: #1199

**Comment by sungam3r** [comment]
@gabrielemilan Thanks for contribution. Replaced by #1199.

**Pull request closed without merging**